File: back-end-java/uiadmin-spring-boot/summer-core/src/main/java/com/jiangruyi/summer/core/util/xybuilder/XyBuilderForm.java
/**
 * 表单构建
 */
package com.jiangruyi.summer.core.util.xybuilder;

import java.util.List;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;
import java.io.Serializable;

import com.alibaba.fastjson.JSONArray;
import com.alibaba.fastjson.JSONObject;

/**
 * @author jry
 *
 */
public class XyBuilderForm implements Serializable {
    private static final long serialVersionUID = 1L;

    public Map<String, String> formType = new HashMap<String, String>(){{
        put("tabs", "TABS切换");
        put("hidden", "隐藏元素");
        put("static", "静态文本");
        put("link", "跳转链接");
        put("text", "单行文本");
        put("password", "密码");
        put("url", "URL网址");
        put("email", "邮箱");
        put("date", "日期");
        put("number", "数字");
        put("digit", "浮点型数字");
        put("tel", "手机号");
        put("textarea", "多行文本");
        put("array", "自定义数组");
        put("select", "下拉框");
        put("selects", "下拉框多选");
        put("radio", "单选框");
        put("checkbox", "多选框");
        put("switch", "开关");
        put("slider", "滑块");
        put("tags", "标签");
        put("datepicker", "日期");
        put("timepicker", "时刻");
        put("datetimepicker", "时间");
        put("daterangepicker", "日期区间");
        put("datetimerangepicker", "时间区间");
        put("rate", "星级评分");
        put("cascader", "级联选择");
        put("region", "省市区联动");
        put("colorpicker", "颜色选择器");
        put("image", "单图上传");
        put("imageflex", "单图列表");
        put("images", "多图上传");
        put("file", "单文件上传");
        put("files", "多文件上传");
        put("poster", "分享海报");
        put("selectlist", "列表选择器");
        put("checkboxtree", "树状表格复选");
        put("markdown", "Markdown");
        put("html", "HTML富文本");
        put("tinymce", "TinyMCE富文本");
        put("sku", "商品规格");
        put("fee", "运费模板");
    }};

    // 配置
    private Map<String, Object> config = new HashMap<String, Object>();

    // 表单提交方法
    private String formMethod = "post";

    // 分栏
    private JSONArray formCols = new JSONArray();

    // 分组
    private JSONArray formGroups = new JSONArray();

    // TABS
    private JSONArray formTabs = new JSONArray();

    // 表单项目
    private JSONArray formItems = new JSONArray();

    // 表单验证规则
    private JSONObject formRules = new JSONObject();

    // 表单值合集
    private Object itemValues = new HashMap<String, Object>();

    // 表单值合集
    private Object formValues = new HashMap<String, Object>();

    public XyBuilderForm() {
        this.config.put("continue", false);
        this.config.put("submitApi", "");
        this.config.put("itemDefaultPosition", "");
        this.config.put("submitButtonTitle", "确认");
        this.config.put("cancelButtonTitle", "取消");
        this.config.put("footerButtonLength", "120px");
        this.config.put("labelPosition", "left");
        this.config.put("labelWidth", "100px");
        this.config.put("defaultUploadDriver", "");
        this.config.put("defaultUploadAction", "/v1/core/index/upload/");
        this.config.put("defaultUploadMaxSize", 512);
    }

    // 修改设置
    public Object setConfig(String name, Object value){
        this.config.put(name, value);
        return this;
    }

    // 表单提交方法
    public Object setFormMethod(String method){
        this.formMethod = method;
        return this;
    }

    // 添加分栏
    public Object addFormCol(String name, Map<String,Object> span, List<Object> itemList, Map<String,Object> extra){
        JSONObject col = new JSONObject();
        col.put("name", name);
        col.put("span", span);
        col.put("itemList", itemList);
        col.put("extra", extra);
        this.formCols.add(col);
        return this;
    }

    // 添加分组
    public Object addFormGroup(String name,String title, List<Object> itemList, HashMap<String,Object> extra){
        JSONObject col = new JSONObject();
        col.put("name", name);
        col.put("title", title);
        col.put("itemList", itemList);
        col.put("extra", extra);
        this.formGroups.add(col);
        return this;
    }

    // 添加表单项目
    public Object addFormItem(String name, String title, String type, Object value){
        JSONObject col = new JSONObject();
        col.put("name", name);
        col.put("title", title);
        col.put("type", type);
        col.put("value", value);
        col.put("extra", new HashMap<>());
        this.formItems.add(col);
        return this;
    }

    // 添加表单项目
    public Object addFormItem(String name,String title, String type, Object value, Map<String, Object> extra){
        JSONObject col = new JSONObject();
        col.put("name", name);
        col.put("title", title);
        col.put("type", type);
        col.put("value", value);
        col.put("extra", extra);
        this.formItems.add(col);
        return this;
    }

    // 添加表单验证规则
    public Object addFormRule(String name, ArrayList<Object> rule){
        this.formRules.put(name, rule);
        return this;
    }

    // 设置表单值
    public Object setFormValues(Object data){
        this.itemValues = data;
        return this;
    }

    // 获取数据
	public Map<String, Object> getData() {
        HashMap<String, Object> data = new HashMap<String, Object>() {
            {
                put("alertList", new HashMap<String, Object>() {
                    {
                        put("top", new ArrayList<Object>());
                        put("bottom", new ArrayList<Object>());
                    }
                });
                put("formMethod", formMethod); 
                put("formCols", formCols);
                put("formGroups", formGroups);
                put("formTabs", formTabs);
                put("formItems", formItems);
                put("formRules", formRules);
                put("formValues", formValues);
                put("itemValues", itemValues);
                put("config", config);
            }
        };
		return data;
	}
}

File: front-end/uiadmin-uniapp/uni_modules/uiadmin-core/uniCloud/cloudfunctions/common/uiadmin-common/util/builder/XyBuilderForm.js
const createConfig = require('uni-config-center')
const shareConfig = createConfig({ // 获取配置实例
    pluginId: 'uiadmin' ,// common/uni-config-center下的插件配置目录名
	defaultConfig: { // 默认配置
		defaultUploadSpaceId: '',
		defaultUploadClientSecret: '',
	}
})
const config = shareConfig.config() 

class XyBuilderForm {

    constructor() {
        this.data = {}
    }

    $formType = {
        'tabs': 'TABS切换',
        'hidden': '隐藏元素',
        'static': '静态文本',
        'link': '跳转链接',
        'text': '单行文本',
        'password': '密码',
        'url': 'URL网址',
        'email': '邮箱',
        'date': '日期',
        'number': '数字',
        'digit': '浮点型数字',
        'tel': '手机号',
        'textarea': '多行文本',
        'array': '自定义数组',
        'select': '下拉框',
        'selects': '下拉框多选',
        'radio': '单选框',
        'checkbox': '多选框',
        'switch': '开关',
        'slider': '滑块',
        'tags': '标签',
        'datepicker': '日期',
        'timepicker': '时刻',
        'datetimepicker': '时间',
        'daterangepicker': '日期区间',
        'datetimerangepicker': '时间区间',
        'rate': '星级评分',
        'cascader': '级联选择',
        'region': '省市区联动',
        'colorpicker': '颜色选择器',
        'image': '单图上传',
        'imageflex': '单图列表',
        'images': '多图上传',
        'file': '单文件上传',
        'files': '多文件上传',
        'poster': '分享海报',
        'selectlist': '列表选择器',
        'checkboxtree': '树状表格复选',
        'markdown': 'Markdown',
        'html': '富文本',
        'tinymce': 'TinyMCE富文本',
        'sku': '商品规格',
        'fee': '运费模板'
    };

    /**
     * 初始化
     * @author jry <ijry@qq.com>
     */
    init() {
        this.data = {
            'alertList': {
                'top': [],
                'bottom': []
            },
            'formMethod': 'post',
            'formCols': [], // 表单项目分栏
            'formGroups': [], // 表单项目分组
            'formItems': [],
            'formValues': [],
            'formRules': [],
            'formTabs': [],
            'config': {
                'continue': false, // 显示继续添加
                'itemDefaultPosition': '',
                'submitButtonTitle': '确认',
                'cancelButtonTitle': '取消',
                'footerButtonLength': '120px',
                'labelPosition': 'left',
                'labelWidth': '100px',
                'defaultUploadDriver': "uniCloud",
				'defaultUploadSpaceId': config.defaultUploadSpaceId,
				'defaultUploadClientSecret': config.defaultUploadClientSecret,
                'defaultUploadAction': '/api/v1/core/upload/upload',
                'defaultUploadMaxSize': 512
            }
        };
        return this;
    }

    /**
     * 设置配置
     * @author jry <ijry@qq.com>
     */
    setConfig($name, $value) {
        this.data['config'][$name] = $value;
        return this;
    }

    /**
     * 添加顶部提醒
     * @author jry <ijry@qq.com>
     */
    addAlertItem($layer, $item) {
        this.data['alertList'][$layer].push($item);
        return this;
    }

    /**
     * 设置tab
     * @author jry <ijry@qq.com>
     */
    addFormTab($tab) {
        this.data['formTabs'].push($tab);
        return this;
    }

    /**
     * 设置提交方法
     * @author jry <ijry@qq.com>
     */
    setFormMethod($method = 'post') {
        this.data['formMethod'] = $method;
        return this;
    }

    /**
     * 构造表单项
     * @author jry <ijry@qq.com>
     */
    getFormItem(
        $name,
        $title,
        $type = 'text',
        $value = '' ,
        $extra = []
    ) {
        let $item = {};
        $item['name'] = $name;
        $item['title'] = $title;
        $item['type'] = $type;
        $item['value'] = $value;
        $item['extra'] = $extra;
        return $item;
    }
    
    /**
     * 添加表单分组
     * @author jry <ijry@qq.com>
     */
    addFormCol(
        $name,
        $span = [],
        $itemList = [],
        $extra = []
    ) {
        this.data['formCols'].push({
            'name': $name,
            'span': $span,
            'itemList': $itemList,
            'extra': $extra
        });
        return this;
    }

    /**
     * 添加表单分组
     * @author jry <ijry@qq.com>
     */
    addFormGroup(
        $name,
        $title,
        $itemList = [],
        $extra = []
    ) {
        this.data['formGroups'].push({
            'name': $name,
            'title': $title,
            'itemList': $itemList,
            'extra': $extra
        });
        return this;
    }

    /**
     * 添加表单项目
     * @author jry <ijry@qq.com>
     */
    addFormItem(
        $name,
        $title,
        $type = 'text',
        $value = '' ,
        $extra = []
    ) {
        let $item = this.getFormItem($name, $title, $type, $value, $extra);
        if ((this.data['formTabs']).length > 0) {
            this.data['formTabs'][(this.data['formTabs']).length - 1]['formItems'].push($item);
            this.data['formTabs'][(this.data['formTabs']).length - 1]['formRules']['default'] = [];
        } else {
            this.data['formItems'].push($item);
            // this.data['formRules']['default'] = [];
        }
        return this;
    }

    /**
     * 添加表单验证
     * @author jry <ijry@qq.com>
     */
    addFormRule($name, $rule){
        if ((this.data['formTabs']).length > 0) {
            this.data['formTabs'][(this.data['formTabs']).length - 1]['formRules'][$name] = $rule;
        } else {
            this.data['formRules'][$name] = $rule;
        }
        return this;
    }

    /**
     * 设置表单数据
     * @author jry <ijry@qq.com>
     */
    setFormValues($data = []) {
        this.data['itemValues'] = $data;
        return this;
    }

    /**
     * 返回数据
     * @author jry <ijry@qq.com>
     */
   getData() {
        return this.data;
    }
}

module.exports = XyBuilderForm
